server/src/main/scala/org/broadinstitute/dsp/workbench/welder/server/ObjectService.scala

```diff
@@ -82,103 +82,112 @@ class ObjectService(
     res.compile.drain
   }
 
-  private def retrieveGcsMetadata(bucketName: GcsBucketName, blobName: GcsBlobName, traceId: TraceId): IO[Option[GcsMetadata]] = for {
+  private def retrieveGcsMetadata(localPath: java.nio.file.Path, bucketName: GcsBucketName, blobName: GcsBlobName, traceId: TraceId): IO[Option[GcsMetadata]] = for {
```

Comment:
the number of arguments and length of the method seems to indicate a violation of the single responsibility principle (in this case both the processing and packaging of the data). Consider making a helper method